.ci/Jenkinsfile

```
@@ -137,20 +149,63 @@ pipeline {
       options { skipDefaultCheckout() }
       when {
         beforeAgent true
-        tag pattern: 'v\\d+\\.\\d+.*', comparator: 'REGEXP'
+        allOf {
+          tag pattern: 'v\\d+\\.\\d+.*', comparator: 'REGEXP'
+          expression { isInternalCI() }
+        }
       }
-      agent { label 'macosx' }
+      environment {
+        BUCKET_NAME = 'internal-ci-artifacts'
+        BUCKET_SUBFOLDER = "${env.REPO}/${env.TAG_NAME}/${env.BUILD_ID}"
+        BUCKET_PATH = "gs://${env.BUCKET_NAME}/${env.BUCKET_SUBFOLDER}"
+        BUCKET_CREDENTIALS = 'internal-ci-gcs-plugin'
+        SIGNED_ARTIFACTS = 'signed-artifacts'
+        BUCKET_SUBFOLDER_SIGNED_ARTIFACTS = "${env.BUCKET_SUBFOLDER}/${env.SIGNED_ARTIFACTS}"
+        BUCKET_SIGNED_ARTIFACTS_PATH = "gs://${env.BUCKET_NAME}/${env.BUCKET_SUBFOLDER_SIGNED_ARTIFACTS}"
+        DIST_FOLDER = 'dist'
+        RELEASE_URL_MESSAGE = "(<https://github.com/elastic/${env.REPO}/releases/tag/${env.TAG_NAME}|${env.TAG_NAME}>)"
+      }
+      agent { label 'macos-code-signer' }
```

`git fetch` gets stalled in one of the CI workers, so let's use the one we know it works

```suggestion
      // There are some environmental issues with some of the CI workers
      // let's use the one we know it works.      
      //agent { label 'macos-code-signer' }      
      agent { label 'worker-c07yc0cejyvy' }
```